For reference, before the fix, the test fails:
```
(...)
time="2026-04-13T09:19:32Z" level=info msg="Created pod in provider"
time="2026-04-13T09:19:32Z" level=info msg="receive GetPod \"nginx\""
time="2026-04-13T09:19:32Z" level=debug msg="Pod nginx exists, updating pod in provider"
time="2026-04-13T09:19:32Z" level=info msg="receive UpdatePod \"nginx\""
time="2026-04-13T09:19:32Z" level=info msg="Updated pod in provider"
time="2026-04-13T09:19:32Z" level=info msg="receive GetPod \"nginx\""
time="2026-04-13T09:19:32Z" level=info msg="receive CreatePod \"nginx\""
time="2026-04-13T09:19:32Z" level=info msg="Created pod in provider"
time="2026-04-13T09:19:32Z" level=info msg="receive GetPod \"nginx\""
time="2026-04-13T09:19:32Z" level=debug msg="Updated pod status in kubernetes" new phase= new reason= old phase= old reason=
time="2026-04-13T09:19:32Z" level=debug msg="Updated pod status in kubernetes" new phase= new reason= old phase= old reason=
time="2026-04-13T09:19:33Z" level=debug msg="Updated pod status in kubernetes" new phase=Running new reason= old phase= old reason=
--- FAIL: TestUpdatePodStatusUsesKubernetesResourceVersion (0.10s)
    pod_test.go:415: assertion failed: 0 (gotResourceVersion string) != 123 (podFromKubernetes.ResourceVersion string)
--- FAIL: TestUpdatePodStatusRejectsZeroResourceVersion (0.10s)
    pod_test.go:482: assertion failed: error while updating pod status in kubernetes: Pod "nginx" is invalid: metadata.resourceVersion: Invalid value: "0": must be specified for an update (type *errors.withStack) is not nil: expected no error: fix at node/pod.go:252 should copy the Kubernetes ResourceVersion onto the provider pod before UpdateStatus
time="2026-04-13T09:19:33Z" level=warning msg="There was a conflict, maybe trying to delete a Pod that has been recreated: Operation cannot be fulfilled on pods \"nginx\": test conflict"
time="2026-04-13T09:19:33Z" level=warning msg="Not deleting pod because pods \"nginx\" not found"
time="2026-04-13T09:19:33Z" level=warning msg="Not deleting pod because remote pod has different UID" k8sPodUID=bbbbb uid=aaaaa
time="2026-04-13T09:19:33Z" level=debug msg="Cleaning up pod from deletion cache" key=default/any-pod
```

Please, @Blackmamba23 add the fix.

--- a/node/pod.go
+++ b/node/pod.go
@@ -249,7 +249,7 @@ func (pc *PodController) updatePodStatus(ctx context.Context, podFromKubernetes 
 	// We need to do this because the other parts of the pod can be updated elsewhere. Since we're only updating
 	// the pod status, and we should be the sole writers of the pod status, set the current ResourceVersion to
 	// satisfy optimistic concurrency requirements.
-	podFromProvider.ResourceVersion = "0"
+	podFromProvider.ResourceVersion = podFromKubernetes.ResourceVersion
 	if _, err := pc.client.Pods(podFromKubernetes.Namespace).UpdateStatus(ctx, podFromProvider, metav1.UpdateOptions{}); err != nil && !errors.IsNotFound(err) {
 		span.SetStatus(err)
 		return pkgerrors.Wrap(err, "error while updating pod status in kubernetes")

